core/src/main/java/org/apache/carbondata/core/datamap/TableDataMap.java

```diff
@@ -80,10 +83,11 @@
   /**
    * It is called to initialize and load the required table datamap metadata.
    */
-  TableDataMap(AbsoluteTableIdentifier identifier, DataMapSchema dataMapSchema,
+  TableDataMap(CarbonTable table, DataMapSchema dataMapSchema,
```

Comment:
How could you make sure carbontable is updated here whenever any changes happens to schema of table if you pass it to the constructor. Because TableDataMap is cached and it would not be created every time

Reply:
when the schema is updated. So im checking for columns and then removing the datamap, and creates new datamap for the latest schema, so during that the carbon table im passing will be updated

Reply:
I don't think it is required to drop complete datamap as only new segments will be having latest schema.